section/algorithms.html

```diff
@@ -1,539 +1,686 @@
-      </li>
-      <li>
-        If |authServer| [=url/scheme=] is not `https`, return failure.
-      </li>
-      <li>
-        <p>Construct a {{ GrantRequest }} |grantRequest: GrantRequest| as follows:</p>
+    <h3>Resource server operations (RS)</h3>
+    <section>
+      <h4>Create incoming payment</h4>
+      <p>The create incoming payment request refers
+        to the <a data-cite="oprs/operations/create-incoming-payment#">Open Payments create incoming payment</a>.</p>
+      <div class="algorithm">
+        When asked to <dfn>send a create incoming payment request</dfn>,
+        given |websiteWalletAddressDetails:WalletAddressDetails|, |amountValue:DOMString|,
+        |incomingPaymentGrant:Grant|, perform the following steps.
         <ol>
           <li>
-            Set |grantRequest|'s {{GrantRequest/client}} to |userWalletDetails|.{{WalletAddressDetails/id}}.
+            Let |resourceServer:URL| be the result of running [=URL parser=] with |websiteWalletAddressDetails|'s {{WalletAddressDetails/resourceServer}} property.
+          </li>
+          <li>
+            If |resourceServer| [=url/scheme=] is not `https`, return failure.
           </li>
           <li>
-            Let |incomingPaymentAccess| be a new {{ Access }} object with:
+            Let |incomingPaymentsUrl:URL| be the result of running [=URL parser=] with the string formed by concatenating |resourceServer| and "/incoming-payments".
+          </li>
+          <li>
+            <p>Let |incomingAmount:Amount| be a new {{Amount}} with:</p>
             <ul>
-              <li>{{Access/type}} set to `"incoming-payment"`</li>
-              <li>{{Access/actions}} set to `["create"]`</li>
-              <li>{{Access/identifier}} set to |websiteWalletAddressDetails|'s {{WalletAddressDetails/id}} property</li>
+              <li>{{Amount/value}} set to |amountValue|</li>
+              <li>{{Amount/assetCode}} set to |websiteWalletAddressDetails|'s {{WalletAddressDetails/assetCode}}</li>
+              <li>{{Amount/assetScale}} set to |websiteWalletAddressDetails|'s {{WalletAddressDetails/assetScale}}</li>
             </ul>
           </li>
           <li>
-            Set |grantRequest|'s {{GrantRequest/access_token}} to a new {{ RequestAccessToken }} with:
+            Let |expiresAt:DOMString| be the current date-time plus 10 minutes, formatted as an ISO 8601 string.
+            <p class="note">The <code>expiresAt</code> field is set to the current date plus 10 minutes, as users typically spend an average of 5 minutes on a website, making it unnecessary to keep the session active longer. Ideally, the browser should terminate the session when the user navigates away from the site. However, if that doesn’t happen for any reason, the expiration date will ensure the session is ended. If the session expires before the user leaves the website, the browser should reinitialize the session.</p>
```

Comment:
The "typically spend an average of 5 minutes on a website" is _citation needed_ material. If no references for this data, it's acceptable to call it an assumption we make for this purpose.